tests/qemu-tests/tests/user/test_aarch64_example.py

```diff
@@ -0,0 +1,136 @@
+from __future__ import annotations
+
+import gdb
+from capstone.arm64_const import ARM64_INS_BL
+
+import pwndbg.gdblib.disasm
+import pwndbg.gdblib.nearpc
+from pwndbg.gdblib.disasm.instruction import InstructionCondition
+
+AARCH64_GRACEFUL_EXIT = """
+mov x0, 0
+mov x8, 93
+svc 0
+"""
+
+SIMPLE_FUNCTION = f"""
+
+bl my_function
+b end
+
+my_function:
+    ret
+
+end:
+{AARCH64_GRACEFUL_EXIT}
+"""
+
+
+def test_syscall_annotation(qemu_assembly_run):
+    """ """
+    qemu_assembly_run(AARCH64_GRACEFUL_EXIT, "aarch64")
+
+    instructions = pwndbg.gdblib.disasm.near(
+        address=pwndbg.gdblib.regs.pc, instructions=3, emulate=True
+    )[0]
+    future_syscall_ins = instructions[2]
+
+    assert future_syscall_ins.syscall == 93
+    assert future_syscall_ins.syscall_name == "exit"
+
+    gdb.execute("stepuntilasm svc")
+
+    # Both for emulation and non-emulation, ensure a syscall at current PC gets enriched
+    instructions = pwndbg.gdblib.disasm.emulate_one(), pwndbg.gdblib.disasm.no_emulate_one()
+
+    for i in instructions:
+        assert i.syscall == 93
+        assert i.syscall_name == "exit"
```

Comment:
Hmm eventually, we want the context display to look correctly; wouldn't it make more sense to assert on context display?

Those tests are good so far; its just that a context display check would correspond to the exact thing user expects to work properly; while current asserts may still pass if context would not display the enriched syscalls.

Reply:
That's a good call - future tests will focus on what the user sees 👍